R/rnet_junctions.R

```diff
@@ -0,0 +1,30 @@
+#' Get points where 3 or more linestrings meet (junctions on road networks)
+#'
+#' @inheritParams rnet_breakup_vertices
+#' @export
+#' @examples
+#' has_sfheaders <- requireNamespace("sfheaders", quietly = TRUE)
+#' if(has_sfheaders) {
+#' rnet <- rnet_roundabout
+#' j <- rnet_junction(rnet)
+#' plot(rnet$geometry)
+#' plot(j, add = TRUE)
+#' }
+rnet_junction <- function(rnet) {
+  stopifnot(requireNamespace("sfheaders", quietly = TRUE))
+  coordinates <- sfheaders::sf_to_df(rnet)
+  # names(coordinates)
+  # head(coordinates)
+  coordinates <- as.matrix(coordinates[c("x", "y", "linestring_id")])
+  L1_index <- ncol(coordinates)
```

Comment:
I used the term `L1_index` since that's the same name returned by `sf::st_coordinates`. Maybe you can change it to `linestring_index` (or something similar). Moreover, I think that the code now works only with objects without Z or M dimensions (but that's not a big deal).

Reply:
Would be good if it could work with z coords too. `linestring_id` is the name returned, so think that's right (for xy points at least), no?

```
> rnet <- rnet_roundabout
> j <- rnet_junction(rnet)
>   coordinates <- sfheaders::sf_to_df(rnet)
>   names(coordinates)
[1] "sfg_id"        "linestring_id" "x"             "y"    
```

Reply:
TBH I'm not sure. Tomorrow I will check an example with XYZ and XYZM dimensions.